adapters/repos/db/migrator.go

```diff
@@ -253,6 +253,63 @@ func (m *Migrator) RecalculateVectorDimensions(ctx context.Context) error {
 	return nil
 }
 
+func (m *Migrator) RecountProperties(ctx context.Context) error {
+	count := 0
+	m.logger.
+		WithField("action", "recount").
+		Info("Recounting properties, this may take a while")
+
+	clearedShard := map[string]bool{}
+
+	// Iterate over all indexes
+	for _, index := range m.db.indices {
+		// Iterate over all shards
+		err := index.IterateObjects(ctx, func(index *Index, shard *Shard, object *storobj.Object) error {
+			if _, ok := clearedShard[shard.name]; !ok {
+				shard.propLengths.Clear()
+				clearedShard[shard.name] = true
+			}
+			count = count + 1
+			props, _, err := shard.analyzeObject(object)
+			if err != nil {
+				m.logger.WithField("error", err).Error("could not analyze object")
+				return nil
+			}
+
+			if err := shard.addPropLengths(props); err != nil {
+				m.logger.WithField("error", err).Error("could not add prop lengths")
+				return nil
+			}
+
+			shard.propLengths.Flush(false)
+
+			return nil
+		})
+		if err != nil {
+			m.logger.WithField("error", err).Error("could not iterate over objects")
+		}
+
+		// Flush the propLengths to disk
+		err = index.IterateShards(ctx, func(index *Index, shard *Shard) error {
+			return shard.propLengths.Flush(false)
+		})
+		if err != nil {
+			m.logger.WithField("error", err).Error("could not flush prop lengths")
+		}
+
+	}
+	go func() {
+		for {
+			m.logger.
+				WithField("action", "recount").
+				Warnf("Recounted %v objects. Recounting properties complete. Please remove environment variable 	RECOUNT_PROPERTIES_AT_STARTUP before next startup", count)
+			time.Sleep(5 * time.Minute)
+		}
+	}()
+
+	return nil
```

Comment:
what happens if this crashes during runtime?

Reply:
Nothing gets flushed to disk, so it is the same as if it didn't run at all